src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -130,13 +130,14 @@ def add_latest_version(
         url_to_artifact: dict[str, set]
             Used to detect artifacts that have similar repos.
         """
-        if item["url"] == "" and item["version"] != "unspecified":
+        if item["url"] == "" and item["version"] != "unspecified" and item["group"] and item["name"]:
             gav = f"{item['group']}:{item['name']}:{item['version']}"
             urls = find_repo(gav, ["scm.url"])
-            if len(urls) != 0:
+            # TODO decide how to handle multiple returned URLs
+            url = next((item for item in urls), "")
```

Comment:
Because `urls` is already an `Iterator` instance, we could use it directly in `next()`: `url = next(urls, "")`
Reference: https://docs.python.org/3/library/functions.html#next